packages/jsxstyle-utils/src/getStyleCache.ts

```diff
@@ -2,6 +2,20 @@ import getStyleKeysForProps from './getStyleKeysForProps';
 import addStyleToHead from './addStyleToHead';
 import stringHash from './stringHash';
 
+export interface InsertRuleCallback {
+  (rule: string, props?: {}): boolean | void;
+}
+
+export interface StyleCache {
+  reset(): void;
+  getClassName(props: { [key: string]: any }, classNameProp?: string): string;
```

Comment:
In my codebases, I like to have a `Dict` type:

```ts
export type Dict<T> = {
  [key: string]: T,
};
```
which makes this a little easier to express:

```ts
  getClassName(props: Dict<any>, classNameProp?: string): string;
```

Reply:
ooh, this is neat.